fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/Ordering.java

```diff
@@ -93,50 +169,70 @@ public class Ordering {
      * contradiction causing a predicate to always evaluate to {@code false}. In other cases, we may encounter
      * {@code x = 5} and {@code x = $p} where it is unclear if we just encountered a contradiction as well or
      * if the predicate is just redundant (where {@code $p} is bound to {@code 5} when the query is executed).
+     * {@link SetOperationsOrdering}s.
      */
     @Nonnull
-    private final SetMultimap<Value, Comparison> equalityBoundValueMap;
+    private final SetMultimap<Value, Binding> bindingMap;
 
     /**
-     * A list of {@link OrderingPart}s where none of the contained expressions is equality-bound. This list
-     * defines the actual order of records.
+     * A {@link PartiallyOrderedSet} of {@link Value}s.
      */
     @Nonnull
-    private final PartiallyOrderedSet<OrderingPart> orderingSet;
+    private final PartiallyOrderedSet<Value> orderingSet;
 
+    /**
+     * Indicator if the records flowed are to be considered distinct, thus producing a strict order. This field
+     * should get deprecated as it is not correct to assign this property to all enumerated orderings. For instance,
+     * an ordering that produces {@code a, b, c} is also ordered by {@code a, b}. While the former one might be strict,
+     * the latter one may not. I think for now we should just interpret this indicator in a way that only enumerated
+     * orderings that contain all values of the ordering are strict if this indicator is {@code true}.
```

Comment:
Do we ever have cause to enumerate a _longer_ orderings? With `a, b, c` distinct, `a, b, c, d` is also available.

Reply:
No, we do not consider that. The reason is that strictly ordering should be ordered as a separate property but is not (yet)! In the future we can utilize that to also enumerate what you are suggesting. Also, I think only the next key can be added.